sdk/core/Azure.Core/src/Shared/RetriableStream.cs

```diff
@@ -151,15 +146,8 @@ public override int Read(byte[] buffer, int offset, int count)
             }
 
             public override bool CanRead => _currentStream.CanRead;
-            public override bool CanSeek { get; }
-            public override long Length
-            {
-                get
-                {
-                    _lengthException?.Throw();
-                    return _length;
-                }
-            }
+            public override bool CanSeek { get; } = false;
+            public override long Length => _length ?? throw new NotSupportedException();
```

Comment:
wouldn't this be odd that "CanSeek" is always false but "Length" is sometimes available? I.e. should we delegate CanSeek to wrapped stream? (and what do we do with Seek? would default impl throw?)

Reply:
The retrieable stream is more complicated than just wrapping a stream and implementing Seek would require additional logic to figure out if we need to request another segment etc. So we can't support the CanSeek right now and the `Length` is the best effort.